src/show.jl

```diff
@@ -0,0 +1,174 @@
+function Base.show(io::IO, fid::File)
```

Comment:
It would be nice if there was an easy way to print the file mode `"r", "r+", "w"` etc... Not sure if there's an easy way to go about this since it doesn't seem to be captured in the File struct

Reply:
Without storing the original mode choice, we can't have a distinction between, e.g. `"cw", "w", "r+"`, but we can get read-write vs read only:
```julia
julia> hfile = h5open("/tmp/test.h5", swmr = true)
🗂️ HDF5.File: read only (swmr) /tmp/test.h5
└─ 📄 complexf64

julia> close(hfile)

julia> hfile = h5open("/tmp/test.h5")
🗂️ HDF5.File: read only /tmp/test.h5
└─ 📄 complexf64
```

Added in the newest commit.

Reply:
That's awesome. Maybe we can put them all in parens. And also perhaps put them after the filename is printed?  (read only, swmr)  ? 
```
🗂️ HDF5.File: /tmp/test.h5 (read only, swmr)

```

Reply:
I agree with all in parens, but I'm not as sure about after the filename; I've worked/working in an environment where we end up with very long filenames (both deep path and near-maximum file names ...), and to bury some useful information way at the end would be unfortunate.

Reply:
That makes sense. One minor thing, lets also add the dash between `(write|read)-only`